src/code/code_800EC960.c

```diff
@@ -4577,52 +4599,63 @@ void Audio_ClearSariaBgm2(void) {
     sSariaBgmPtr = NULL;
 }
 
-void func_800F5510(u16 seqId) {
-    func_800F5550(seqId);
-    func_800F5E18(SEQ_PLAYER_BGM_MAIN, seqId, 0, 0, 1);
+void Audio_PlayMorningSceneSequence(u16 seqId) {
+    Audio_PlaySceneSequence(seqId);
+    // Writing a value of 1 to ioPort 0 will be used by
+    // `NA_BGM_FIELD_LOGIC` to play `NA_BGM_FIELD_MORNING` first
+    Audio_PlaySequenceWithSeqPlayerIO(SEQ_PLAYER_BGM_MAIN, seqId, 0, 0, 1);
 }
 
-void func_800F5550(u16 seqId) {
-    u8 sp27 = 0;
-    u16 nv;
+void Audio_PlaySceneSequence(u16 seqId) {
+    u8 fadeInDuration = 0;
+    u8 skipHarpIntro;
 
     if (func_800FA0B4(SEQ_PLAYER_BGM_MAIN) != NA_BGM_WINDMILL) {
         if (func_800FA0B4(SEQ_PLAYER_BGM_SUB) == NA_BGM_LONLON) {
+            // Stop sequence now
             func_800F9474(SEQ_PLAYER_BGM_SUB, 0);
+            // Terminate all internal audio cmds
             Audio_QueueCmdS32(0xF8000000, 0);
         }
 
-        if ((sSeqFlags[D_80130630] & SEQ_FLAG_5) && sSeqFlags[(seqId & 0xFF) & 0xFF] & SEQ_FLAG_4) {
-
-            if ((D_8013062C & 0x3F) != 0) {
-                sp27 = 0x1E;
+        if ((sSeqFlags[sPrevSceneSeqId] & SEQ_FLAG_STORE_PREV_SPOT) &&
+            sSeqFlags[seqId & 0xFF & 0xFF] & SEQ_FLAG_STORE_SPOT) {
+            // Start the sequence from the spot where it last left off last time it was played in the scene
+            if ((sSeqSpot & 0x3F) != 0) {
+                fadeInDuration = 30;
             }
 
-            func_800F5E18(SEQ_PLAYER_BGM_MAIN, seqId, sp27, 7, D_8013062C);
+            // Write the sequence spot to start from into ioPort 7
+            Audio_PlaySequenceWithSeqPlayerIO(SEQ_PLAYER_BGM_MAIN, seqId, fadeInDuration, 7, sSeqSpot);
 
-            D_8013062C = 0;
+            sSeqSpot = 0;
         } else {
-            nv = (sSeqFlags[(seqId & 0xFF) & 0xFF] & SEQ_FLAG_6) ? 1 : 0xFF;
-            func_800F5E18(SEQ_PLAYER_BGM_MAIN, seqId, 0, 7, nv);
-            if (!(sSeqFlags[seqId] & SEQ_FLAG_5)) {
-                D_8013062C = 0xC0;
+            // Start the sequence from the beginning
+
+            // Writes to ioPort 7. See `SEQ_FLAG_SKIP_HARP_INTRO` for writing a value of 1 to ioPort 7.
+            // Note: writing 0xFF (-1) to an ioPort will do nothing for all sequences.
```

Comment:
```suggestion
            // Note: writing 0xFF (-1) to an ioPort will do nothing for all vanilla sequences.
```
as in, the sequences don't have a behavior for it, it's not the interpreter code ignoring it?

Reply:
As in sequences are designed to use `-1` as "this is an empty value" through checks such as `>= 0` to do stuff. Also, the `soundScriptIO[]` slots will be set to `-1` when reset. I haven't seen any sequence use `-1` in any way. Does that answer your question?

I'm also not sure what adding "vanilla" does here

Reply:
I mean that if I write my own sequence, it's up to me that writing -1 does nothing, not the game's code?

Reply:
In a sense yes, but it's implied by code in `seqplayer.c` that -1 means to reset the io value, so designing the sequence to use -1 as a meaningful value might cause issues if you don't design the sequence in that way.

Reply:
thanks for clearing that up, then maybe a better suggestion from me could be
```suggestion
            // Note: writing 0xFF (-1) to an ioPort will be ignored by sequences.
```
that to me makes it clear the "do nothing" part happens on the sequences' side

Maybe it would be good to elaborate on why too,
```suggestion
            // Note: writing 0xFF (-1) to an ioPort will be ignored by sequences. -1 is intended as a default (?) to clear (?) the io value
```
(?)

Reply:
I think there's a better place to document the full intent with an io value of `-1` in general. Here, it shouldn't be more than just a quick note that `-1` is the `NONE` of io values

Reply:
Oh I didn't know -1 is "the `NONE` of io values" ( :sunglasses: ), okay if that's something way more general than this single spot this is good

Reply:
I might just drop this comment entirely then and just use `#define SEQ_IO_VAL_NONE -1`. 

Might be annoying though since the ioVal is interpreted as both s8 and u8 depending on code, so it might have to go back and forth between `-1` and `0xFF`, would need to test

Reply:
`(s8/u8/...)SEQ_IO_VAL_NONE` would do the trick wouldn't it?